File: frontend/src/constants/apiPolicyPlatformDefaults.ts
/** Mirrors backend `ApiPolicyPlatformDefaults` — single source for UI "platform defaults" labels. */
export const API_POLICY_PLATFORM_DEFAULTS = {
  outputFormats: ['DOCX', 'PDF'],
  outputModes: ['SYNC_STREAM', 'SYNC_DOWNLOAD_URL', 'ASYNC_TASK'],
  batchEnabled: true,
  syncMaxItems: 100,
  asyncMaxItems: 10_000,
  docxEncryptionEnabled: false,
  pdfEncryptionEnabled: false,
  saveGeneratedDocuments: true,
  invocationRecordRetentionDays: 90,
  documentRetentionDays: 30,
} as const

/** BDD invocation record retention presets (days). */
export const INVOCATION_RECORD_RETENTION_PRESETS = [7, 30, 90, 180, 365, 1095, 2555] as const

/** BDD document retention presets (days) — only when saveGeneratedDocuments is true. */
export const DOCUMENT_RETENTION_PRESETS = [7, 30, 90, 180, 365] as const

export function arraysEqual<T>(left: readonly T[], right: readonly T[]): boolean {
  if (left.length !== right.length) {
    return false
  }
  return left.every((value, index) => value === right[index])
}

export function isOutputPolicyPlatformDefault(
  outputFormats: readonly string[],
  outputModes: readonly string[],
): boolean {
  return (
    arraysEqual(outputFormats, API_POLICY_PLATFORM_DEFAULTS.outputFormats) &&
    arraysEqual(outputModes, API_POLICY_PLATFORM_DEFAULTS.outputModes)
  )
}

export function isBatchLimitsPlatformDefault(
  batchEnabled: boolean,
  syncMaxItems: number,
  asyncMaxItems: number,
): boolean {
  return (
    batchEnabled === API_POLICY_PLATFORM_DEFAULTS.batchEnabled &&
    syncMaxItems === API_POLICY_PLATFORM_DEFAULTS.syncMaxItems &&
    asyncMaxItems === API_POLICY_PLATFORM_DEFAULTS.asyncMaxItems
  )
}

export function isEncryptionPlatformDefault(
  docxEncryptionEnabled: boolean,
  pdfEncryptionEnabled: boolean,
): boolean {
  return (
    docxEncryptionEnabled === API_POLICY_PLATFORM_DEFAULTS.docxEncryptionEnabled &&
    pdfEncryptionEnabled === API_POLICY_PLATFORM_DEFAULTS.pdfEncryptionEnabled
  )
}

File: backend/src/main/java/com/bank/docgen/apimgmt/domain/ApiPolicyPlatformDefaults.java
package com.bank.docgen.apimgmt.domain;

import java.util.List;

/**
 * Single source of truth for platform-wide API policy defaults used when materializing
 * skeleton policies and seeding new rows.
 */
public final class ApiPolicyPlatformDefaults {

    public static final List<String> OUTPUT_FORMATS = List.of("DOCX", "PDF");
    public static final List<String> OUTPUT_MODES = List.of("SYNC_STREAM", "SYNC_DOWNLOAD_URL", "ASYNC_TASK");
    public static final boolean BATCH_ENABLED = true;
    public static final int BATCH_SYNC_MAX_ITEMS = 100;
    public static final int BATCH_ASYNC_MAX_ITEMS = 10000;
    public static final boolean DOCX_ENCRYPTION_ENABLED = false;
    public static final boolean PDF_ENCRYPTION_ENABLED = false;
    public static final boolean SAVE_GENERATED_DOCUMENTS = true;
    public static final int INVOCATION_RECORD_RETENTION_DAYS = 90;
    public static final int DOCUMENT_RETENTION_DAYS = 30;
    public static final String ALLOWED_AD_GROUPS_JSON = "[]";
    public static final String OUTPUT_FORMATS_JSON = "[\"DOCX\",\"PDF\"]";
    public static final String OUTPUT_MODES_JSON = "[\"SYNC_STREAM\",\"SYNC_DOWNLOAD_URL\",\"ASYNC_TASK\"]";

    private ApiPolicyPlatformDefaults() {
    }
}
